app/lib/differential_expression_service.rb

```diff
@@ -184,14 +198,20 @@ def self.run_differential_expression_job(cluster_file, study, user, annotation_n
   def self.validate_annotation(cluster_file, study, annotation_name, annotation_scope)
     cluster = study.cluster_groups.by_name(cluster_file.name)
     raise ArgumentError, "cannot find cluster for #{cluster_file.name}" if cluster.nil?
-
     if DifferentialExpressionResult.where(study: study,
                                           cluster_group: cluster,
                                           annotation_name: annotation_name,
                                           annotation_scope: annotation_scope).exists?
+      # Question: Does each study/cluster_group/annotation_name/annotation_scope combo
+      # map to a single DifferentialExpressionResult? If yes, find_by should be fine
+      # If no, please raise a PR comment to use .where instead
+      result = DifferentialExpressionResult.find_by(study: study,
```

Comment:
Yes - this is correct.  That quaternion (had to look that up!) is used as the uniqueness constraint here: https://github.com/broadinstitute/single_cell_portal_core/blob/development/app/models/differential_expression_result.rb#L29

This means that `find_by` will always return the correct answer as there can only be one.  That being said, I think this whole block starting at the `if` clause can be simplified now to the following:
```suggestion
      result = DifferentialExpressionResult.find_by(study: study,
                                              cluster_group: cluster,
                                              annotation_name: annotation_name,
                                              annotation_scope: annotation_scope)
      if result.present?
        raise ArgumentError,
              "#{annotation_name} already exists for #{study.accession}:#{cluster_file.name}, " \
              "please delete result #{result.id} before retrying"
      end
```

Reply:
`946b49e` Thanks for the simplification - much cleaner!